tsdb/engine/tsm1/cache_map.go

```diff
@@ -0,0 +1,208 @@
+package tsm1
+
+import (
+	"math/rand"
+	"os"
+	"strconv"
+	"sync"
+	"sync/atomic"
+
+	xxhash "github.com/OneOfOne/xxhash/native"
+)
+
+var (
+	bucketIds uint64 = 256 // must be a power of 2
+	seed      uint64
+)
+
+func init() {
+	if s := os.Getenv("CACHE_MAP_BUCKETS"); s != "" {
+		n, err := strconv.Atoi(s)
+		if err != nil {
+			panic(err.Error())
+		}
+		bucketIds = uint64(n)
```

Comment:
Comment above says bucketIds must be a power of two but this doesn't enforce that.